File: packages/appwsgi/appwsgi%201233.tar.bz2/appwsgi/www/appointment/appointment.js
// Copyright(c) gert.cuykens@gmail.com
appointment=
{
 'name':function()
 {
  http.fml=function(v)
  {
   document.getElementById('appointments-dg').innerHTML=''
   appointments=
   {
    'id':'appointments',
    'rec':v['rec'],
    'des':v['des'],
    'onselect':function()
    {
     document.getElementById('uid').value=appointments.selection[0]
     appointment.ca()
    }
   }
   dg.init(appointments)
  }
  http.xml ='{"cmd":"name",\n'
  http.xml+=' "gid":"'+document.getElementById('view').innerHTML+'",\n'
  http.xml+=' "name":"'+document.getElementById('appointment').value+'"}\n'
  http.url ='appointment.wsgi'
  http.send()
 },

 'ca':function()
 {
  http.fml=function(v)
  {
   document.getElementById('appointments-dg').innerHTML=''
   appointments=
   {
    'id':'appointments',
    'rec':v['rec'],
    'des':v['des'],
    'onselect':function()
    {
     document.getElementById('uid').value=appointments.selection[1]
     document.getElementById('aid').value=appointments.selection[2]
     document.getElementById('calendar').value=appointments.selection[3]
     document.getElementById('appointment').value=appointments.selection[4]
     if(appointments.selection[3]=='None')document.getElementById('calendar').value='0000-00-00 00:00:00'
     calendar.display()
    }
   }
   dg.init(appointments)
  }
  http.xml ='{"cmd":"ca",\n'
  http.xml+=' "gid":"'+document.getElementById('view').innerHTML+'",\n'
  http.xml+=' "uid":"'+document.getElementById('uid').value+'",\n'
  http.xml+=' "calendar":"'+document.getElementById('calendar').value+'",\n'
  http.xml+=' "appointment":"'+document.getElementById('appointment').value+'"}\n'
  http.url ='appointment.wsgi'
  http.send()
 },

 'find':function()
 {
  http.fml=function(v)
  {
   document.getElementById('appointments-dg').innerHTML=''
   appointments=
   {
    'id':'appointments',
    'rec':v['rec'],
    'des':v['des'],
    'onselect':function()
    {
     document.getElementById('uid').value=appointments.selection[1]
     document.getElementById('aid').value=appointments.selection[2]
     document.getElementById('calendar').value=appointments.selection[3]
     document.getElementById('appointment').value=appointments.selection[4]
     if(appointments.selection[3]=='None')document.getElementById('calendar').value='0000-00-00 00:00:00'
     calendar.display()
    }
   }
   dg.init(appointments)
  }
  http.xml ='{"cmd":"find",\n'
  http.xml+=' "gid":"'+document.getElementById('view').innerHTML+'",\n'
  http.xml+=' "calendar":"'+document.getElementById('calendar').value+'",\n'
  http.xml+=' "appointment":"'+document.getElementById('appointment').value+'"}\n'
  http.url ='appointment.wsgi'
  http.send()
 },
 
 'save':function()
 {
  var cmd
  if(document.getElementById('aid').value=='new'){cmd='insert'}else{cmd='update'}
  http.fml=function(v)
  {
   document.getElementById('appointments-dg').innerHTML=''
   if(document.getElementById('aid').value=='new')document.getElementById('aid').value=v['rid']
  }
  http.xml ='{"cmd":"'+cmd+'",\n'
  http.xml+=' "gid":"'+document.getElementById('view').innerHTML+'",\n'
  http.xml+=' "aid":"'+document.getElementById('aid').value+'",\n'
  http.xml+=' "uid":"'+document.getElementById('uid').value+'",\n'
  http.xml+=' "calendar":"'+document.getElementById('calendar').value+'",\n'
  http.xml+=' "appointment":"'+document.getElementById('appointment').value+'"}\n'
  http.url ='appointment.wsgi'
  http.send()
 },

 'remove':function()
 {
  http.fml=function(v)
  {
   document.getElementById('aid').value='new'
   document.getElementById('appointment').value=''
   if(c=document.getElementById('appointments'))while(c.lastChild)c.removeChild(c.lastChild)
  }
  http.xml ='{"cmd":"remove",\n'
  http.xml+=' "gid":"'+document.getElementById('view').innerHTML+'",\n'
  http.xml+=' "aid":"'+document.getElementById('aid').value+'"}'
  http.url ='appointment.wsgi'
  if(document.getElementById('aid').value!='new'){http.send()}
 }
}

File: packages/appwsgi/appwsgi%201014.tar.bz2/appwsgi/www/appointment/appointment.py
# Copyright(c) gert.cuykens@gmail.com
from frame import Frame
from session import Session
from db import Db

def application(environ, response):
    cookie = environ.get('HTTP_COOKIE','')
    document = environ['wsgi.input'].read(int(environ.get('CONTENT_LENGTH','0')))
    db = Db()
    v = Frame.read(document)
    s = Session(db,cookie,v['gid'])

    if v['cmd'] == 'overview':
                               db.execute("SELECT calendar,appointment FROM appointments WHERE appointments.calendar >= %s AND appointments.calendar < %s",(v['from'],v['to']))
    elif v['cmd'] == 'insert':      
        if s.GID  == 'admin' : db.execute("INSERT INTO appointments VALUES (%s,0,%s,%s)",(v['uid'],v['calendar'],v['appointment']))
        elif s.GID == 'guest': db.execute("INSERT INTO appointments VALUES (%s,0,%s,%s)",(v['uid'],v['calendar'],v['appointment']))
    elif v['cmd'] == 'update':
        if s.GID  == 'admin':  db.execute("UPDATE appointments SET calendar=%s, appointment=%s WHERE aid=%s AND uid=%s",(v['calendar'],v['appointment'],v['aid'],v['uid']))
        elif s.GID == 'guest': db.execute("UPDATE appointments SET uid=%s, calendar=%s, appointment=%s WHERE aid=%s",(v['uid'],v['calendar'],v['appointment'],v['aid']))
    elif v['cmd'] == 'remove':
        if s.GID  == 'admin':  db.execute("DELETE FROM appointments WHERE aid=%s AND uid=%s",(v['aid'],v['uid']))
        elif s.GID == 'guest': db.execute("DELETE FROM appointments WHERE aid=%s",(v['aid']))
    elif v['cmd'] == 'name':
        if s.GID  == 'admin':  db.execute("SELECT uid,name,adress,city,country,phone FROM users WHERE name LIKE %s AND uid=%s",('%'+v['name']+'%',v['uid']))
        elif s.GID == 'guest': db.execute("SELECT uid,name,adress,city,country,phone FROM users WHERE name LIKE %s",("%"+v['name']+"%"))
    elif v['cmd'] == 'find':
                               db.execute("SELECT users.name, appointments.* FROM users, appointments WHERE users.uid = appointments.uid AND appointments.calendar >= %s AND appointments.appointment LIKE %s GROUP BY appointments.aid",(v['calendar'],"%"+v['appointment']+"%"))
    elif v['cmd'] == 'ca':
                               db.execute("SELECT users.name, appointments.* FROM users, appointments WHERE users.uid = appointments.uid AND appointments.calendar >= %s AND appointments.appointment LIKE %s AND appointments.uid=%s GROUP BY appointments.aid",(v['calendar'],"%"+v['appointment']+"%",v['uid']))

    xml = '<?xml version="1.0" encoding="UTF-8"?>\n'
    xml+= '<root>\n'
    xml+= ' <cmd>'+str(v['cmd'])+'</cmd>\n'
    xml+= ' <sid>'+str(s.SID)+'</sid>\n'
    xml+= ' <exp>'+str(s.EXP)+'</exp>\n'
    xml+= ' <uid>'+str(s.UID)+'</uid>\n'
    xml+= ' <gid>'+str(s.GID)+'</gid>\n'
    xml+= Frame.write(db)
    xml+= '</root>'

    response('200 OK', [('Content-type', 'text/xml'), ('Set-Cookie', s.COOKIE)])
    return [xml]

